prometheus/testutil/testutil.go

```diff
@@ -112,31 +112,43 @@ func ToFloat64(c prometheus.Collector) float64 {
 	panic(fmt.Errorf("collected a non-gauge/counter/untyped metric: %s", pb))
 }
 
-// CollectAndCount collects all Metrics from the provided Collector and returns their number.
-//
-// This can be used to assert the number of metrics collected by a given collector after certain operations.
-//
-// This function is only for testing purposes, and even for testing, other approaches
-// are often more appropriate (see this package's documentation).
-func CollectAndCount(c prometheus.Collector) int {
-	var (
-		mCount int
-		mChan  = make(chan prometheus.Metric)
-		done   = make(chan struct{})
-	)
-
-	go func() {
-		for range mChan {
-			mCount++
-		}
-		close(done)
-	}()
+// CollectAndCount registers the provided Collector with a newly created
+// pedantic Registry. It then calls GatherAndCount with that Registry and with
+// the provided metricNames. In the unlikely case that the registration or the
+// gathering fails, this function panics. (This is inconsistent with the other
+// CollectAnd… functions in this package and has historical reasons. Changing
+// the function signature would be a breaking change and will therefore only
+// happen with the next major version bump.)
+func CollectAndCount(c prometheus.Collector, metricNames ...string) int {
```

Comment:
I totally don't see `CollectAndCount(c prometheus.Collector, metricNames ...string)` signature. It does not make sense to me - too many ways of doing the same thing.

Reply:
I don't understand your comment. What's the other way of doing the same thing?

Reply:
You answered in main thread, I did not know there exists implementations of collector with many names.